### PR TITLE
refactor: telemetry package refactored to modular structure

### PR DESCRIPTION
**Pull request by tphakala**
The telemetry package has been refactored to a modular structure to improve code organization and maintainability. The package now consists of multiple modules, each responsible for a specific aspect of telemetry data collection and processing.

Added metrics collection to MQTT module
Added BirdNET processing time metrics

### CODE REVIEW

File: internal/imageprovider/imageprovider.go [hunk 1]
```diff
@@ -1,95 +1,120 @@
...
 	"unsafe"
 
 	"github.com/tphakala/birdnet-go/internal/telemetry"
+	"github.com/tphakala/birdnet-go/internal/telemetry/metrics"
 )
 
+// ImageProvider defines the interface for fetching bird images.
 type ImageProvider interface {
-	fetch(scientificName string) (BirdImage, error)
+	Fetch(scientificName string) (BirdImage, error)
```
Comment (isZumpo) [inline internal/imageprovider/imageprovider.go:16]: Why is this function made public?

File: internal/imageprovider/imageprovider.go [hunk 2]
```diff
@@ -1,95 +1,120 @@
...
+	c.nonBirdImageProvider = provider
+}
+
+// SetImageProvider allows setting a custom ImageProvider for testing purposes.
+func (c *BirdImageCache) SetImageProvider(provider ImageProvider) {
+	c.birdImageProvider = provider
+}
+
+// initCache initializes a new BirdImageCache with the given ImageProvider.
+func InitCache(e ImageProvider, t *telemetry.Metrics) *BirdImageCache {
```
Comment (isZumpo) [inline internal/imageprovider/imageprovider.go:55]: Same with this one

File: internal/imageprovider/imageprovider.go [hunk 3]
```diff
@@ -1,95 +1,120 @@
...
+		c.metrics.IncrementCacheHits()
 		return birdImage.(BirdImage), nil
 	}
 
-	// Fetch the bird image from the image provider
 	fetchedBirdImage, err := c.fetch(scientificName)
 	if err != nil {
-		// TODO for now store an empty result in the cache to avoid future queries that would fail.
-		// In the future, look at the error and decide if it was caused by networking and is recoverable.
-		// And if it was, do not store the empty result in the cache.
```
Comment (isZumpo) [inline internal/imageprovider/imageprovider.go:81]: Still think this todo is valid and should not be removed

File: internal/imageprovider/imageprovider.go (hunk 1 above)
Comment (tphakala) [reply internal/imageprovider/imageprovider.go:16]: For unit tests to work

File: internal/imageprovider/imageprovider.go (hunk 3 above)
Comment (tphakala) [reply internal/imageprovider/imageprovider.go:81]: Some comments were removed unintentionally, like this

File: internal/imageprovider/imageprovider.go (hunk 1 above)
Comment (isZumpo) [reply internal/imageprovider/imageprovider.go:16]: Can't the unit tests be in the same package as before?

File: internal/imageprovider/wikipedia.go [hunk 4]
```diff
@@ -163,47 +167,40 @@ func (l *wikiMediaProvider) queryAuthorInfo(thumbnailURL string) (*wikiMediaAuth
...
 
 	if len(wikipediaUserLinks) == 1 {
-		// Return the href and inner text of the Wikipedia user link
 		wikipediaLink := wikipediaUserLinks[0]
 		href = extractHref(wikipediaLink)
 		text = extractText(wikipediaLink)
 		return href, text, nil
 	}
 
-	// Check if all the links have the same href value
```
Comment (isZumpo) [inline internal/imageprovider/wikipedia.go:206]: All of these comments might also be useful to detangle the reason for the differerent if cases

File: internal/imageprovider/imageprovider.go (hunk 3 above)
Comment (isZumpo) [reply internal/imageprovider/imageprovider.go:81]: Ah ok I understand :)

**Pull request merged**